test/distributed/pipelining/test_backward.py

```diff
@@ -124,11 +131,18 @@ def test_stage_backward_weight(self, device):
         ref_loss = loss_fn(ref_out, ref_target)
         ref_loss.backward()
 
+        rtol, atol = None, None
+        if self.device_type == "xpu":
```

Comment:
having device specific logic in the tests is not ideal. Kind of curious, is this only hit for this test or should this logic be in the `torch.testing.assert_close()` util method?

Also if accuracy gap only happens for non-deterministic tests, can we just make the test deterministic?

Reply:
@wincent8 Let's try to set the non-deterministic option instead of tolerance.

Reply:
> having device specific logic in the tests is not ideal. Kind of curious, is this only hit for this test or should this logic be in the `torch.testing.assert_close()` util method?
> 
> Also if accuracy gap only happens for non-deterministic tests, can we just make the test deterministic?

TBH, only these cases exhibit accuracy gaps in non-deterministic tests. For all other cases, XPU behaves as expected.